src/pydata_sphinx_theme/theme/pydata_sphinx_theme/sections/header.html

```diff
@@ -1,8 +1,8 @@
 {% if theme_navbar_start or theme_navbar_center or theme_navbar_end or theme_navbar_persistent %}
 <div class="bd-header__inner bd-page-width">
-  <label class="sidebar-toggle primary-toggle" for="__primary">
+  <button class="sidebar-toggle primary-toggle" aria-label="{{ _('Site navigation') }}">
```

Comment:
I chose this ARIA label and the one below because these strings are already in the translation system.